### ISSUE TITLE
Knowledge distillation on regression task

### ISSUE THREAD

**Issue by graycrown** [google-research/google-research#911]
I noticed the conclusion in your paper - "In contrast, Single -> Multi knowledge distillation improves or matches the performance of the other methods on all tasks except STS, the only regression task in GLUE. **We believe distillation does not work well for regression tasks because there is no distribution over classes passed on by the teacher to aid learning**". 

Did you test the knowledge distillation technique on other regression tasks?  E.g., noise reduction.

Looking forward to your reply.

B.R.

**Comment by Ankur-singh** [comment]
Any updates on this? I am trying to use knowledge distillation in a regression problem. But its very difficult to find any paper or project on it.

I tired using `(MSE(student_preds, GT) + MSE(student_preds, teacher_preds))/2` but this doesn't seem to work. Currently, I am planing to do `a * MSE(student_preds, GT) + (1-a) * MSE(student_preds, teacher_preds)`. However, I am not sure whether it will work or not.

**Comment by boredtylin** [comment]
I dont think directly using MSE loss would be an ideal technique. This is problematic when the output of teacher model mismatch the given label. For classification tasks, the logits offer more info than hard one-hot label. but in regression tasks the logits could be erroneous.
